AviRecord.cpp build failure with x265 4.0

**Describe the bug**
Build error with x265 4.0 (worked well with version 3.x)

**To Reproduce**
Build with x265 4.0
Here is the error:
```
/tmp/fceux-20240916-58775-96iqtc/src/drivers/Qt/AviRecord.cpp:463:8: error: no matching function for call to 'x265_encoder_encode'
ret = x265_encoder_encode( hdl, &nal, &i_nal, pic, &pic_out );
^~~~~~~~~~~~~~~~~~~
/opt/homebrew/Cellar/x265/4.0/include/x265.h:2496:5: note: candidate function not viable: no known conversion from 'x265_picture *' to 'x265_picture **' for 5th argument
int x265_encoder_encode(x265_encoder *encoder, x265_nal **pp_nal, uint32_t *pi_nal, x265_picture *pic_in, x265_picture **pic_out);
^
/tmp/fceux-20240916-58775-96iqtc/src/drivers/Qt/AviRecord.cpp:496:12: error: no matching function for call to 'x265_encoder_encode'
ret = x265_encoder_encode( hdl, &nal, &i_nal, NULL, &pic_out );
^~~~~~~~~~~~~~~~~~~
/opt/homebrew/Cellar/x265/4.0/include/x265.h:2496:5: note: candidate function not viable: no known conversion from 'x265_picture *' to 'x265_picture **' for 5th argument
int x265_encoder_encode(x265_encoder *encoder, x265_nal **pp_nal, uint32_t *pi_nal, x265_picture *pic_in, x265_picture **pic_out);
^
``` 

The change is in https://bitbucket.org/multicoreware/x265_git/commits/c69c113960834400545bc4bce2830ff51dcb86b3, the 5th argument changed from x265_picture *pic_out to x265_picture **pic_out

**Expected behavior**
No build error

**Screenshots**
Not applicable

**(please complete the following information):**
Not applicable

**Additional context**
See https://github.com/Homebrew/homebrew-core/pull/189134#issuecomment-2357564913

--- a/src/drivers/Qt/AviRecord.cpp
+++ b/src/drivers/Qt/AviRecord.cpp
@@ -460,7 +460,14 @@ static int encode_frame( unsigned char *inBuf, int width, int height )
 	pic->stride[1] = width/2;
 	pic->stride[2] = width/2;
 
+#ifdef MAX_SCALABLE_LAYERS
+	/* Handle API changes for scalable layers output in x265 4.0 */
+	x265_picture *pics[MAX_SCALABLE_LAYERS] = {NULL};
+	pics[0] = pic;
+	ret = x265_encoder_encode( hdl, &nal, &i_nal, pic, pics );
+#else
 	ret = x265_encoder_encode( hdl, &nal, &i_nal, pic, &pic_out );
+#endif
 
 	if ( ret <= 0 )
 	{
@@ -493,7 +500,14 @@ static int close(void)
 	/* Flush delayed frames */
 	while( hdl != NULL )
 	{
+#ifdef MAX_SCALABLE_LAYERS
+	    /* Handle API changes for scalable layers output in x265 4.0 */
+	    x265_picture *pics[MAX_SCALABLE_LAYERS] = {NULL};
+	    pics[0] = pic;
+	    ret = x265_encoder_encode( hdl, &nal, &i_nal, pic, pics );
+#else
 	    ret = x265_encoder_encode( hdl, &nal, &i_nal, NULL, &pic_out );
+#endif
 
 	    if ( ret <= 0 )
 	    {

